lib/Serialization/DeserializationErrors.h

```diff
@@ -359,82 +429,111 @@ class OverrideError : public llvm::ErrorInfo<OverrideError,
   }
 };
 
-class TypeError : public llvm::ErrorInfo<TypeError, DeclDeserializationError> {
+// Service class for errors with an underlying cause.
+class CausedError {
```

Comment:
maybe ErrorCause?

Reply:
Changed it to `ErrorWithUnderlyingReason`.